Vagrantfile

```diff
@@ -0,0 +1,100 @@
+  # `vagrant box outdated`. This is not recommended.
+  # config.vm.box_check_update = false
+
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine. In the example below,
+  # accessing "localhost:8080" will access port 80 on the guest machine.
+  # NOTE: This will enable public access to the opened port
+  config.vm.network "forwarded_port", guest: 5000, host: 5000
+
+  # Create a forwarded port mapping which allows access to a specific port
+  # within the machine from a port on the host machine and only allow access
+  # via 127.0.0.1 to disable public access
+  # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"
+
+  # Create a private network, which allows host-only access to the machine
+  # using a specific IP.
+  # config.vm.network "private_network", ip: "192.168.33.10"
+
+  # Create a public network, which generally matched to bridged network.
+  # Bridged networks make the machine appear as another physical device on
+  # your network.
+  # config.vm.network "public_network"
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "../data", "/vagrant_data"
+
+  # Provider-specific configuration so you can fine-tune various
+  # backing providers for Vagrant. These expose provider-specific options.
+  # Example for VirtualBox:
+  #
+  # config.vm.provider "virtualbox" do |vb|
+  #   # Display the VirtualBox GUI when booting the machine
+  #   vb.gui = true
+  #
+  #   # Customize the amount of memory on the VM:
+  #   vb.memory = "1024"
+  # end
+  #
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    sudo apt-get update
+    sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+xz-utils tk-dev libffi-dev liblzma-dev python-openssl git gunicorn
+    [[ -d ~/.pyenv ]] && git -C ~/.pyenv pull || git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    [[ -f ~/.pyprofile ]] && rm -f ~/.pyprofile 
+    touch ~/.pyprofile
+    grep -qF -- 'source ~/.pyprofile' ~/.profile || echo 'source ~/.pyprofile' >> ~/.profile
+    PYPROFILE='export PYENV_ROOT="$HOME/.pyenv"\nexport PATH="$PYENV_ROOT/bin:$PATH"\nexport PATH="~/.local/bin:$PATH"'
+    grep -qF --  "$PYPROFILE" ~/.pyprofile || echo  "$PYPROFILE" >> ~/.pyprofile
```

Comment:
What you've done works but in case you misunderstand why:

the `\n`s on line 75 are being converted to new line characters by Vagrant's interpretation of this script.

If you ran the two commands as written `PYPROFILE='...\n...\n...'` and `echo "$PYPROFILE" >> ~/.pyprofile` on command line, you would end up with literally `\n` in the file rather than a new line.

Normally you'd expect the environment variable to have \n rather than a new line and you should do `echo -e` to make sure they get converted to newlines (like you have for the echo pyenv init line)

Reply:
to ensure things were not being duplicated when using only provision i this helped me solve by only creating when not exists